DFReader.py

```diff
@@ -176,6 +176,9 @@ def set_mult_ids(self, mult_ids, mult_lookup):
                     self.units[i] = MULT_TO_PREFIX[unitmult]+self.units[i]
                 else:
                     self.units[i] = "%.4g %s" % (unitmult, self.units[i])
+                if self.units[i] in FORMAT_TO_STRUCT:
```

Comment:
I think that the "if self.units[i] in FORMAT_TO_STRUCT" bit is working because the prefix for "centi" and the code for "int16_t * 100" are both 'c' - but in my mind they aren't really the same 'c'!

Assuming your aim is that any parameter which has no-unit but does have a multiplier specified, has that multiplier applied to the value, instead of becoming a unit prefix, then maybe something like this is cleaner:
```suggestion
                # 
                if self.units[i] == "":
                    self.msg_mults[i] = mult_lookup[mult_ids[i]]
                elif unitmult in MULT_TO_PREFIX:
                    self.units[i] = MULT_TO_PREFIX[unitmult]+self.units[i]
                else:
                    self.units[i] = "%.4g %s" % (unitmult, self.units[i])
```
(to replace all lines 175 -181)

Not checked in detail, but I think this should then work for any multiplier which is set on a unitless parameter, even if that multiplier is not 100.